tests/UpdatePaymentMethod/UpdatePaymentMethodWithoutAddingToBalanceTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+
+namespace Laravel\Cashier\Tests\UpdatePaymentMethod;
+
+use Illuminate\Support\Facades\Event;
+use Laravel\Cashier\Events\MandateUpdated;
+use Laravel\Cashier\FirstPayment\Actions\AddBalance;
+use Laravel\Cashier\FirstPayment\Actions\AddGenericOrderItem;
+use Laravel\Cashier\FirstPayment\FirstPaymentHandler;
+use Laravel\Cashier\Order\Order;
+use Laravel\Cashier\Tests\BaseTestCase;
+use Laravel\Cashier\Tests\Fixtures\User;
+use Mollie\Api\MollieApiClient;
+use Mollie\Api\Resources\Payment;
+
+class UpdatePaymentMethodWithoutAddingToBalanceTest extends BaseTestCase
+{
+    /** @test */
+    public function canUpdatePaymentMethodWithoutAddingToBalance()
+    {
+        $this->withPackageMigrations();
+        Event::fake();
+
+        $payment = $this->getMandatePaymentStub();
+
+        $owner = factory(User::class)->create([
+            'id' => $payment->metadata->owner->id,
+            'mollie_customer_id' => 'cst_unique_customer_id',
+        ]);
+
+        $handler = new FirstPaymentHandler($payment);
+
+        $this->assertTrue($owner->is($handler->getOwner()));
+
+        $actions = $handler->getActions();
+        $this->assertCount(1, $actions);
+
+        $firstAction = $actions[0];
+        $this->assertInstanceOf(AddBalance::class, $firstAction);
+        $this->assertMoneyEURCents(1000, $firstAction->getTotal());
+        $this->assertEquals('Test add balance 1', $firstAction->getDescription());
+
+        $this->assertFalse($owner->hasCredit());
+        $this->assertNull($owner->mollie_mandate_id);
+
+        $this->assertEquals(0, $owner->orderItems()->count());
+        $this->assertEquals(0, $owner->orders()->count());
+
+        $order = $handler->execute();
+
+        $owner = $owner->fresh();
+
+        $this->assertTrue($owner->hasCredit());
+        $credit = $owner->credit('EUR');
+        $this->assertMoneyEURCents(1000, $credit->money());
+
+        $this->assertEquals(1, $owner->orderItems()->count());
+        $this->assertEquals(1, $owner->orders()->count());
+
+        $this->assertInstanceOf(Order::class, $order);
+        $this->assertTrue($order->isProcessed());
+
+        $this->assertEquals(1, $order->items()->count());
+
+        $this->assertNotNull($owner->mollie_mandate_id);
+        $this->assertEquals($payment->mandateId, $owner->mollie_mandate_id);
+
+        Event::assertDispatched(MandateUpdated::class, function (MandateUpdated $e) use ($owner, $payment) {
+            $this->assertTrue($e->owner->is($owner));
+            $this->assertSame($e->payment->id, $payment->id);
+
+            return true;
+        });
```

Comment:
The actual test doesn't start until line 75.

Line 25 (`$payment = $this->getMandatePaymentStub();`) until line 74 are all operations to get into a base state to test against, overcomplicating this test.

Running the payment method only requires this initial state (perhaps a bit more, but not much more):

- billable.mollie_customer_id is known
- billable.mollie_mandate_id contains a mandate id that is about to be replaced
- billable has 0 credits
- billable has no orders or orderItems

Can you simplify it so we can still understand it a few months from now? :)

Reply:
refactor tests

Reply:
I simplified the tests

Reply:
I simplified the tests